### DISCUSSION TITLE
Custom XCom backend leads to error

### DISCUSSION THREAD

**Discussion by Pocoder** [apache/airflow#21426] [General] [answered]
### Apache Airflow version

2.2.3 (latest released)

### What happened

I've created my own XCom Backend 'S3XComBackend' with no additional logic in folder include/custom_xcom_backend.py and then tried to configure my docker-compose file to work with it. I added `AIRFLOW__CORE__XCOM_BACKEND: 'include.custom_xcom_backend.S3XComBackend'` to environments and `- ./include:/opt/airflow/include` to volumes. Then I ran docker-compose up and faced ERROR:

> airflow_postgres_1 is up-to-date
> airflow_redis_1 is up-to-date
> Starting airflow_airflow-init_1 ... done
> 
> ERROR: for airflow-webserver  Container "a3c1a81acdcd" exited with code 1.
> 
> ERROR: for airflow-worker  Container "a3c1a81acdcd" exited with code 1.
> 
> ERROR: for airflow-scheduler  Container "a3c1a81acdcd" exited with code 1.
> 
> ERROR: for airflow-triggerer  Container "a3c1a81acdcd" exited with code 1.
> 
> ERROR: for flower  Container "a3c1a81acdcd" exited with code 1.
> ERROR: Encountered errors while bringing up the project.

### How to reproduce

This is my xcom_backend:
``` python
from typing import Any
import pickle
import uuid

from airflow.models.xcom import BaseXCom
from airflow.providers.amazon.aws.hooks.s3 import S3Hook


class S3XComBackend(BaseXCom):

    @staticmethod
    def serialize_value(value: Any):
        return BaseXCom.serialize_value(value)

    @staticmethod
    def deserialize_value(result) -> Any:
        return BaseXCom.deserialize_value(result)
```

### Operating System

mac-os

### Versions of Apache Airflow Providers

```
apache-airflow==2.2.3
apache-airflow-providers-amazon==2.6.0
apache-airflow-providers-ftp==2.0.1
apache-airflow-providers-http==2.0.2
apache-airflow-providers-imap==2.1.0
apache-airflow-providers-sqlite==2.0.1

```

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
PLease avoid creating github issue if you have trouble with making your custom things to work - Github Issues is not to help you debug your problems. Use discussions for this where you might get help from others. 

But in order to do that, you need to share logs and errors and tell what you tried to do to investigate the issue. Then you MIGHT get help if you show that you tried. There is no chance somoene here will debug your problems here for you before you've shown you made your effort and give enough information that might give others a chance to help you.

**Comment by Pocoder** [comment, marked as answer]
Solved my issue by adding volumes in airflow-init section in docker-compose.

**Reply by spb722** [reply]
thanks , worked for me too

**Reply by wkrill** [reply]
I still get your error after changing
```
mkdir -p /sources/logs /sources/dags /sources/plugins
chown -R "${AIRFLOW_UID}:0" /sources/{logs,dags,plugins}
```
to 
```
mkdir -p /sources/logs /sources/dags /sources/plugins /sources/include
chown -R "${AIRFLOW_UID}:0" /sources/{logs,dags,plugins,include}
```
and leave 
```
    volumes:
      - .:/sources
```
in the airflow-init section in docker-compose. 
Did you add volumens in another way? Thank you for posting this issue.
